infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -304,11 +333,21 @@ private void completeSnapshotApply(LogReplicationEntry inputEntry) {
         logEntrySinkBufferManager = new LogEntrySinkBufferManager(ackCycleTime, ackCycleCnt, bufferSize,
                 logReplicationMetadataManager.getLastProcessedLogTimestamp(), this);
 
-
         log.info("Sink manager completed SNAPSHOT transfer for {} and has transit to {} state.",
                 inputEntry, rxState);
     }
 
+    /**
+     * Get a snapshot transfer end marker:
+     * 1. Mark the metadata that snapshot transfer phase is done.
+     * 2. Generate an APPLY_SNAPSHOT_EVENT and submit to executor
+     * @param message
+     */
+    private void processSnapshotTransferEndMarker(LogReplicationEntry message) {
+        snapshotWriter.snapshotTransferDone(message);
+        completeSnapshotApply(message);
```

Comment:
is this complete or startSnapshotApply? it seems the semantic of that method changed, we should also revisit the messages that are within that method.

Reply:
Also that method is setting the snapshotApplied in the logReplicationMetadataManager and its just started, right?

Reply:
The actual apply is still not happening on this dedicated thread but on the receive method itself.

Reply:
Rename the function for the confusion.